src/FakeItEasy/A.cs

```diff
@@ -44,7 +44,7 @@ public static T Fake<T>(Action<IFakeOptions<T>> optionsBuilder)
         {
             Guard.AgainstNull(optionsBuilder, nameof(optionsBuilder));
 
-            return (T)FakeAndDummyManager.CreateFake(typeof(T), options => optionsBuilder((IFakeOptions<T>)options));
+            return (T)FakeAndDummyManager.CreateFake(new DummyCreationSession(), typeof(T), options => optionsBuilder((IFakeOptions<T>)options));

```

Comment:
Well, it would make sense if we injected *all* the dependencies at the static entry points, but in this case it looks a bit inconsistent. I think it'd be cleaner to inject it lower.